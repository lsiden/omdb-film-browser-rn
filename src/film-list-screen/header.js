import React from "react"
import PropTypes from "prop-types"
import { Text, View, StyleSheet } from "react-native"
import { connect } from "react-redux"

import SearchBar from "components/search-bar"

const style = StyleSheet.create({
  wrapper: {
    borderStyle: "solid",
    borderBottomColor: "black",
    borderBottomWidth: 1
  },
  message: {
    fontSize: 14
  },
  messageWrapper: {
    paddingTop: 2,
    paddingBottom: 2,
    paddingLeft: 20
  }
})

export const header = ({ totalResults, query }) => {
  const msg = query ? `Found ${totalResults} results.` : "Type part of a title."
  return (
    <View style={style.wrapper}>
      <SearchBar />
      <View style={style.messageWrapper}>
        <Text style={style.message}>{msg}</Text>
      </View>
    </View>
  )
}

header.propTypes = {
  totalResults: PropTypes.number.isRequired,
  query: PropTypes.string
}

export default connect(state => ({
  query: state.query
}))(header)
