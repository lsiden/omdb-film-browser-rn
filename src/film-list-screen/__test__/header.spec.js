import React from "react"
import renderer from "react-native-test-utils"

import { header as Header } from "../header"

jest.mock("components/search-bar", () => "SearchBar")

const defaultProps = () => {
  return {
    totalResults: 99
  }
}

const createWrapper = (props = {}) =>
  renderer(<Header {...{ ...defaultProps(), ...props }} />)

it("renders", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
