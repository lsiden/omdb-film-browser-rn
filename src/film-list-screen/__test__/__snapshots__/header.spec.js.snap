// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "borderStyle": "solid",
    }
  }
>
  <SearchBar />
  <View
    style={
      Object {
        "paddingBottom": 2,
        "paddingLeft": 20,
        "paddingTop": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Type part of a title.
    </Text>
  </View>
</View>
`;
